File: api/src/main/java/com/cai/funtour/pojo/Result.java
package com.cai.funtour.pojo;


import lombok.Getter;
import lombok.Setter;
import lombok.ToString;

import java.io.Serializable;
import java.util.HashMap;
import java.util.Map;

/**
 * @author ：caizhiyuan
 * @date ：Created in 2022/4/21 16:29
 * @description：返回结果
 */

@Getter
@Setter
@ToString
public class Result implements Serializable {
    private Integer code;
    private String errMes;
    private Object data;

    public Result() {
    }

    public Result(Integer code, String errMes) {
        this.code = code;
        this.errMes = errMes;
    }

    public Result(Integer code, String errMes, Object data) {
        this.code = code;
        this.errMes = errMes;
        this.data = data;
    }

    public static Result ok(){
        return new Result(200,"",null);
    }

    public static Result error(String mes){
        return new Result(500,mes,null);
    }

    public static Result error(Integer code, String mes){
        return new Result(code,mes,null);
    }

    public static Result toData(Map data){
        return new Result(200,"",data);
    }

    public static Result toData(String key, Object value){
        HashMap<String, Object> data = new HashMap<String, Object>();
        data.put(key, value);
        return new Result(200,"",data);
    }

    public static Result toData(Object value){
        return new Result(200,"",value);
    }
}

File: model/result.go
package model

import (
	"encoding/json"
	"fmt"
	"funtour/tool"
	hessian "github.com/apache/dubbo-go-hessian2"
)

type Result struct {
	Code   int32  `json:"code"`
	ErrMes string `json:"errMes"`
	Data   string `json:"data"`
}

type Data struct {
	result string `json:"data"`
}

func (d *Data) JavaClassName() string {
	return "java.lang.Object"
}

func init() {
	hessian.RegisterPOJO(&Result{})
}

//hessian序列化实现接口
func (u *Result) JavaClassName() string {
	return "com.cai.funtour.pojo.Result"
}

func Ok() *Result {
	r := &Result{
		Code:   200,
		ErrMes: "",
		Data:   "",
	}
	tool.Info("封装结果:", r)
	return r
}

func Error(code int32, errorMes string) *Result {
	r := &Result{
		Code:   code,
		ErrMes: errorMes,
		Data:   "",
	}
	tool.Info("封装结果:", r)
	return r
}

func ToData(result interface{}) *Result {
	var data string
	switch result.(type) {
	case string:
		data = fmt.Sprintln(result)
	default:
		result, _ := json.Marshal(&result)
		data = string(result)
	}

	r := &Result{
		Code:   200,
		ErrMes: "",
		Data:   data,
	}
	tool.Info("封装结果:", result)
	return r
}

func ToDataWithName(name string, result interface{}) *Result {
	mapData := make(map[string]*interface{})
	mapData[name] = &result
	data, _ := json.Marshal(&mapData)
	r := &Result{
		Code:   200,
		ErrMes: "",
		Data:   string(data),
	}
	tool.Info("封装结果:", result)
	return r
}